### ISSUE TITLE
Caching methods with array parameter

### ISSUE THREAD

**Issue by tobiasb** [cordialgerm/FluentCache#1]
I am caching a method with one parameter which is a `string[]`. What I found was that the key didn't seem to be derived from the array's actual content. I wrote a quick test and confirmed:

```
            _sut.Calculate(new[] { "a", "b", "c" });
            _sut.Calculate(new[] { "d", "e", "f" });

            A.CallTo(() => _fakeInnerService.Calculate(A<string[]>._))
             .MustHaveHappened(Repeated.Exactly.Twice); //This fails
```

The second call to `Calculate` did not invoke the inner method but returned the result from the first call. The only way to work around that is to create a new type to pass to `Calculate` holding the string array with a `ToString` override doing a `string.Join(",", array)`.

Am I missing something here or is this intended?

**Comment by evan-rash** [comment]
The default `ICache.GetParameterCacheKeyValue` implementation just returns `parameterValue.ToString()`.

You can fix this by overriding `GetParameterCacheKeyValue` in your `ICache` implementation to generate a CSV representation of the array with `string.Join(",", array)`.

In the next release I will also update the default GetParameterCacheKeyValue implementation to handle this situation

**Comment by evan-rash** [comment]
Resolved with 3.0.0.2
